File: php/features.php
<?php
declare(strict_types=1);

// WahlumfragenDatenbank SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class WahlumfragenDatenbankFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new WahlumfragenDatenbankBaseFeature();
            case "test":
                return new WahlumfragenDatenbankTestFeature();
            default:
                return new WahlumfragenDatenbankBaseFeature();
        }
    }
}

File: rb/features.rb
# WahlumfragenDatenbank SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module WahlumfragenDatenbankFeatures
  def self.make_feature(name)
    case name
    when "base"
      WahlumfragenDatenbankBaseFeature.new
    when "ratelimit"
      WahlumfragenDatenbankRatelimitFeature.new
    when "retry"
      WahlumfragenDatenbankRetryFeature.new
    when "test"
      WahlumfragenDatenbankTestFeature.new
    when "timeout"
      WahlumfragenDatenbankTimeoutFeature.new
    else
      WahlumfragenDatenbankBaseFeature.new
    end
  end
end
